### PR TITLE
Update 16-progressive-web-apps.mdx

### PR DESCRIPTION
**Pull request by preitinger**
The code in the original doc is not valid typescript because window.PushSubscription is mixed up with webpush.PushSubscription from 'web-push'. I checked in possible fixes (and old code commented out- should be removed when accepted in main branch).

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 687438dd94c6e108de646fb0a7db189b218e7fbf

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: docs/02-app/01-building-your-application/07-configuring/16-progressive-web-apps.mdx [hunk 1]
```diff
@@ -437,14 +443,31 @@ export async function sendNotification(message: string) {
...
+          p256dh: subscription.keys?.p256dh ?? ''
+        }
+      },
       JSON.stringify({
         title: 'Test Notification',
         body: message,
         icon: '/icon.png',
       })
     )
+    // old, but a type bug because PushSubscription from client is mixed up with webpush.PushSubscription which is required by webpush.sendNotification():
```
Comment (ijjk) [inline docs/02-app/01-building-your-application/07-configuring/16-progressive-web-apps.mdx:461]: If this is invalid we should remove it and not leave it around

File: docs/02-app/01-building-your-application/07-configuring/16-progressive-web-apps.mdx [hunk 2]
```diff
@@ -415,9 +418,12 @@ webpush.setVapidDetails(
   process.env.VAPID_PRIVATE_KEY!
 )
 
-let subscription: PushSubscription | null = null
+// TODO The following old type must be changed because it is a read-only type and PushSubscription cannot be passed as an arg for a server function because it is not a plain object.
```
Comment (ijjk) [inline docs/02-app/01-building-your-application/07-configuring/16-progressive-web-apps.mdx:421]: Same as https://github.com/vercel/next.js/pull/70400/files#r1774077881

File: docs/02-app/01-building-your-application/07-configuring/16-progressive-web-apps.mdx [hunk 3]
```diff
@@ -251,7 +251,10 @@ function PushNotificationManager() {
       ),
     });
     setSubscription(sub);
-    await subscribeUser(sub);
+    // TODO The following old line is a bug because subscribeUser is a server function and may only receive plain objects
```
Comment (ijjk) [inline docs/02-app/01-building-your-application/07-configuring/16-progressive-web-apps.mdx:254]: Same as https://github.com/vercel/next.js/pull/70400/files#r1774077881

File: docs/02-app/01-building-your-application/07-configuring/16-progressive-web-apps.mdx (hunk 1 above)
Comment (preitinger) [reply docs/02-app/01-building-your-application/07-configuring/16-progressive-web-apps.mdx:461]: Yes in the final merge - absolutely. I thought this is a first proposal from myself to explain the problem. And therefore I wanted the old code to be visible. But true one can always see the diff...

**Comment by leerob** [comment]
Could you please open a new PR once it's ready to be reviewed? 🙏 Thank you!

**Comment by preitinger** [comment]
I am sorry I did not have the time to provide a complete ready-to-go fix. Actually, it is not so important, to have examples that are perfectly runnable, right? It's more about seeing how one can start...

**Pull request closed without merging**